bench/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+if(NOT NATS_BUILD_LIB_STATIC)
+  MESSAGE(FATAL_ERROR
+    "Building tests require static library, or run CMake with -DBUILD_TESTING=OFF")
+  return()
+endif()
+
+include_directories(${PROJECT_SOURCE_DIR}/src)
+include_directories(${PROJECT_SOURCE_DIR}/bench)
+
+if(NATS_BUILD_WITH_TLS)
+  include_directories(${OPENSSL_INCLUDE_DIR})
+endif(NATS_BUILD_WITH_TLS)
+if(NATS_BUILD_LIBUV_EXAMPLE AND LIBUV_DIR)
```

Comment:
Do you plan to have bench for libuv/libevent and streaming? If not, you could remove those for now and add back only if needed?

Reply:
I'll remove what I can without breaking my default dev build settings. (That's the reason I have stan - I normally build/test with it ON locally, and `natsp.h` won't compile otherwise.

1/5 I just cut&pasted from test, but prob. better to consolidate into a test.h and test/bench/... to reuse the makefile and macros. A separate PR though? This chain is already complicated as it is.

Reply:
What I meant is that since it was a new directory with a separate make file and there was - at the moment - a single file that is neither STAN or libuv/libevent, there was no need for those "if" build statements. But I think I understand: if we have stan/libuv/libevent on, and although the bench directory does not have any related test, some of the includes would cause build issues maybe? It seems that you have removed it in the last commit, so it seems to be ok now?
But agree if you prefer to do in different PR, etc..

Reply:
I merged bench->test to avoid all this duplication.